dgbpy/bokeh_apps/trainui/main.py

```diff
@@ -407,18 +408,27 @@ def setIterProgress(msgstr):
   def setProgressComplete(msgstr):
     progress['Ended'] = True
 
+  def setFold(msgstr):
+    if progress['state'] == ProgState.Ready:
+      progress['doCrossVal'] = True #allow crossval ui from training platform callbacks
+    progress['ifold'], progress['n_folds'] = uibokeh.getProgMsg(msgstr)
+
   def resetProgressDict():
-    progress['iter'], progress['n_iters'] = 0,0
+    progress['iter'], progress['n_iters'] = 0, 0
     progress['epoch'] = 0
     progress['state'], progress['Ended'] = ProgState.Ready, False
     progress['after_iter'] = False
-    if progress['_chunkTemp'] == progress['n_chunks']:
+    resetFold = False
+    if progress['_foldTemp'] == progress['n_folds']:
+      resetFold, progress['doCrossVal'] = True, False
+      progress['ifold'], progress['n_folds'], progress['_foldTemp'] = 0, 0, 0
+    if progress['_chunkTemp'] == progress['n_chunks'] and resetFold:
       progress['ichunk'], progress['_chunkTemp'], progress['n_chunks'] = 0, 0, 0
       progress['n_epochs'] = 0
     else:
       progress['state'] = ProgState.Running
 
-  def progressMonitorCB(chunk, master, child):
+  def progressMonitorCB(chunk, fold, master, child):
```

Comment:
please do not use the 'master' word in any code. Replace by something adequate, either 'main' or 'parent'